### ISSUE TITLE
Continuously deploy tagged releases

### ISSUE THREAD

**Issue by jtpio** [pyodide/pyodide#1727]
For now there seems to be two ways for loading Pyodide from a CDN:

- using a release: https://cdn.jsdelivr.net/pyodide/v0.17.0/full/pyodide.js
- the latest dev: https://cdn.jsdelivr.net/pyodide/dev/full/pyodide.js

It would be great if there was a way to use intermediate versions, and would help downstream projects progressively update.

Looks like the logic for uploading the assets is in:

https://github.com/pyodide/pyodide/blob/f13f3340e0458bb78ff653383352b61a57cb6c17/.circleci/config.yml#L287-L293

Maybe there could be an additional step that would upload to `s3://pyodide-cdn2.iodide.io/<hash>/full/`, for example `s3://pyodide-cdn2.iodide.io/c8436c3/full/` that would correspond to this commit: c8436c33a7fbee13e1ded97c0bbdaa7d635f2745

**Comment by jtpio** [comment]
As an example, JupyterLite is updating to the latest Pyodide in https://github.com/jupyterlite/jupyterlite/pull/274, and could already start shipping with it. Staying with `dev` is a bit risky as their might still be some API changes happening before the new release.

**Comment by rth** [comment]
> Staying with dev is a bit risky as their might still be some API changes happening before the new release.

Yeah, dev should definitely not be used for production. Even worse you can potentially end up with inconsistent files depending what's loaded from cache (including CDN cache) and what's loaded from the updated origin.

One solution is to tag alpha releases more often, we can probably make one this week, we were planning to.

In the alternatives,
 - deploying for each commit: one distribution is around 180 MB compressed (much more uncompressed) and we have around 40 commit per month on main, that's  7.2GB per month or 86GB per year. I'm not sure that would be very sustainable (or at least that the use case would be worth it).
 - we could potentially make daily or weekly releases in a cron job.

Generally, in terms of infrastructure the issue for us is more the download bandwidth cost rather than size on s3. If there is potentially a commit per day from which users might load Pyodide, this means that CDN caching will be less efficient. 

Another point is that once we release a npm package, it would need to be in compatible with the deployed packages (which will not necessarily be the case for daily/weekly releases).

So overall as a first approach I think I would still be +1 for making alpha releases more often instead of continuous deployment.

**Comment by hoodmane** [comment]
What about weekly releases in a cron job?

**Comment by jtpio** [comment]
More frequent (or weekly) alpha releases sounds good :+1:

**Comment by hoodmane** [comment]
My issue with manual alpha releases is that we will probably just forget to do it.

**Comment by rth** [comment]
We could go with say automated weekly releases in addition to alphas, but maybe not in a cron job. Since our CI is still flaky it's very likely a given job won't deploy. So maybe instead for each commit on master check if something like `/cd/<version>-<year>w<week-number>` exists on S3 and if  not upload the built artifacts there.  I would rather put them in a separate subfolder than the regular releases. But we should make it very clear in the docs that regular releases should be used for productions, that these are dev deployment that may be broken and that this  CDN path might stop working in the future without warning (well not in near future).

We still need to find a way gracefully shut down https://pyodide-cdn2.iodide.io  CDN endpoint, so overall the less long term support we promise for non critical features, the better.
